lib/SPIRV/libSPIRV/SPIRVEntry.h

```diff
@@ -642,7 +642,9 @@ class SPIRVExecutionMode : public SPIRVAnnotation<OpExecutionMode> {
   SPIRVExecutionModeKind getExecutionMode() const { return ExecMode; }
   const std::vector<SPIRVWord> &getLiterals() const { return WordLiterals; }
   SPIRVCapVec getRequiredCapability() const override {
-    return getCapability(ExecMode);
+    auto Caps = getAllEnablingCapabilities(ExecMode);
+    Module->chooseBestCapability(Caps);
```

Comment:
```suggestion
    Module->chooseBestCapability(getAllEnablingCapabilities(ExecMode));
```

Reply:
There is "return Caps" one line below, so we cannot optimize it this way